app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :username, presence: true
```

Comment:
Consider adding validation for email as well.

Reply:
according to https://github.com/plataformatec/devise/blob/master/lib/devise/models/validatable.rb#L31
isn't it enough?

Reply:
Cool, git catch!